eclipse-tractusx/bpdm
# fix: Pool Business Partner Request Validation

## Description

This pull request fixes behaviour of the validation for  create and update business partner requests. In case of missing technical keys the pool would respond with a 400 error for the whole request instead of just rejecting erroneous requests. Now, the pool lists all errors of those requests but still processes requests that have no errors.

## Test Instructions

Try to create and update business partners with requests that have errors as listed in the org/eclipse/tractusx/bpdm/pool/api/model/response/PoolErrorCodes.kt. The POST/PUT endpoints for creating/updating the business partners should list all errors that the requests contain and those requests should be ignored.

After fixing all the errors the create/update requests should be processed, so new business partners should be created/updated.

Fixes #336
